src/simulation_strategies/FixedLossSimulationStrategy.py

```diff
@@ -31,10 +33,13 @@ def __prepare_initial_state(self, input_state: ndarray) -> ndarray:
         """
             This method is used to prepare psi_0 state (formula 23 from ref. [1]).
             :param input_state: Initial lossy bosonic state.
-            :return:
+            :return: Returns initial state of the formula (being an equal superposition
+            on n photons 'smeared' on first n modes).
         """
-        initial_number_of_photons = sum(input_state)
-        prepared_state = input_state[:]
+        initial_number_of_photons = int(sum(input_state))
+        prepared_state = [1 for _ in range(initial_number_of_photons)]
+        while len(prepared_state) < self.number_of_observed_modes:
+            prepared_state.append(0)
         prepared_state /= sqrt(initial_number_of_photons)
```

Comment:
How does this line work? `prepared_state` is of type `list`, am I missing something?

Reply:
prepared_state should be a list of floats, that is correct. And /= operator shouldn't work in this scenario, but yet this code didn't pose any problems during the tests.

I've added casting to array type, so that /= is expected to work now.

Reply:
If this code didn't pose any problems in the tests, it would seem that this code wasn't tested. How about not adding the casting, but making sure that this code is properly called first?

Reply:
I'll look into that.

Reply:
On my tests state, this code IS tested and produces expected results. From list of ints:
`[1, 1, 1, 1, 0]`
it produces
`[0.5, 0.5, 0.5, 0.5, 0]`.
It's called as expected and when expected - I've checked using debugger. Now the question is if this works, should I change it?

On the other hand, I've tried 
`s = [1.0, 1.0, 1.0, 1.0]
s /= 2.0`
which raises unsupported operand type TypeError... When I use
`s = asarray([1.0, 1.0, 1.0, 1.0])
s /= 2.0`
it's all cool. I'm confused. :/

Reply:
I have an answer for you. You use `sqrt` but not `builtin` but the `numpy` version. It returns `numpy.float64` that, during division, considers `list` to be an `array` and returns `numpy.array` as a result. So the thing that lost us here is the `sqrt` not being the default (returning `float`) but from `numpy` (returning trickery).

Reply:
I've added a comment near this line. Do you think that is enough?